diff -urNp -x '*.orig' wl-6.30.223.271.org/src/shared/linux_osl.c wl-6.30.223.271/src/shared/linux_osl.c
--- wl-6.30.223.271.org/src/shared/linux_osl.c	2021-02-25 20:01:38.664328984 +0100
+++ wl-6.30.223.271/src/shared/linux_osl.c	2021-02-25 20:01:38.797662636 +0100
@@ -1063,7 +1063,11 @@ osl_os_get_image_block(char *buf, int le
 	if (!image)
 		return 0;
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 14, 0)
+	rdlen = kernel_read(fp, buf, len, &fp->f_pos);
+#else
 	rdlen = kernel_read(fp, fp->f_pos, buf, len);
+#endif
 	if (rdlen > 0)
 		fp->f_pos += rdlen;
 
