--- wl-6.30.223.248/src/include/linux_osl.h.orig	2014-06-26 12:42:08.000000000 +0200
+++ wl-6.30.223.248/src/include/linux_osl.h	2015-12-05 10:54:47.671456752 +0100
@@ -268,9 +268,6 @@
 #define OSL_PREF_RANGE_LD(va, sz)
 #define OSL_PREF_RANGE_ST(va, sz)
 
-#define OSL_GETCYCLES(x)	((x) = osl_getcycles())
-extern uint osl_getcycles(void);
-
 #define	BUSPROBE(val, addr)	osl_busprobe(&(val), (addr))
 extern int osl_busprobe(uint32 *val, uint32 addr);
 
--- wl-6.30.223.248/src/shared/linux_osl.c.orig	2014-06-26 12:42:08.000000000 +0200
+++ wl-6.30.223.248/src/shared/linux_osl.c	2015-12-05 10:54:59.058199062 +0100
@@ -926,19 +926,6 @@
 	return ((void*)va);
 }
 
-uint
-osl_getcycles(void)
-{
-	uint cycles;
-
-#if defined(__i386__)
-	rdtscl(cycles);
-#else
-	cycles = 0;
-#endif 
-	return cycles;
-}
-
 void *
 osl_reg_map(uint32 pa, uint size)
 {
